types/jest/index.d.ts

```diff
@@ -163,6 +165,10 @@ declare namespace jest {
      * Mocks a module with an auto-mocked version when it is being required.
      */
     function doMock(moduleName: string, factory?: () => unknown, options?: MockOptions): typeof jest;
+    /**
+     * Mocks a module with a provided version when it is being required.
+     */
+    function doMock<T extends any>(moduleName: string, factory: MockModuleFactory<T>, options?: MockOptions): typeof jest;
```

Comment:
I don’t think we should add any more violations of the [`no‑unnecesary‑generics`](https://github.com/microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md) rule.

Reply:
Are you referring to the fact that it is making use of a generic only once?
Without a generic, I don't believe we can safely ensure that the factory function is returning replacements of the correct types in a nice and verbose manner. The `any` approach (which is essentially what we already have) is less than ideal.

Workaround I currently have is to have a function declared before the `jest.mock` with an explicit return type:
```
const replacementFactory = (): Partial<MyModuleType> => {
   return {
       //... replacement values
   }
};

jest.mock("myModule", replacementFactory);
```
Was hoping the generic approach would provide a nice, readable and imho sensible alternative.

Any suggestions or is this a no-go?